backend/src/main/java/com/summerHack/diningTogether/service/UserService.java

```diff
@@ -75,16 +79,23 @@ public UserDetails getUserDetailsByUsername(String username) throws UsernameNotF
      * @throws UserAlreadyExistException User with the username or email exist
      */
     @Transactional
-    public User registerUser(RegisterInput input) throws UserAlreadyExistException {
+    public User registerUser(RegisterInput input, String siteURL) throws UserAlreadyExistException, MessagingException {
         if (userRepository.findByUsername(input.getUsername()).isPresent()
             || userRepository.findByEmail(input.getEmail()).isPresent()) {
             throw new UserAlreadyExistException();
         }
 
+
+
         final User user = mapper.map(input, User.class);
         user.setPassword(passwordEncoder.encode(input.getPassword()));
         user.setCurrency(properties.getDefaultCurrency());
-        return userRepository.save(user);
+        String randomCode = RandomString.make(64);
```

Comment:
You can use UUID for tokens: `UUID.randomUUID().toString()` so you don't have to introduce third party dependencies.